src/library/tabledelegates/keydelegate.cpp

```diff
@@ -0,0 +1,64 @@
+#include "library/tabledelegates/keydelegate.h"
+
+#include <QPainter>
+#include <QStyle>
+#include <QTableView>
+
+#include "moc_keydelegate.cpp"
+#include "track/keyutils.h"
+
+void KeyDelegate::paintItem(
+        QPainter* painter,
+        const QStyleOptionViewItem& option,
+        const QModelIndex& index) const {
+    paintItemBackground(painter, option, index);
+
+    mixxx::track::io::key::ChromaticKey key =
+            index.data().value<mixxx::track::io::key::ChromaticKey>();
+    // Get the key text with the user-provided notation
+    bool keyColorsEnabled = index.data(Qt::UserRole).value<bool>();
+    int rectWidth = 0; // only leave space for rect if it is rendered
+    if (keyColorsEnabled) {
+        const QColor keyColor = KeyUtils::keyToColor(key);
+        rectWidth = 8; // 4px width + 4px right padding
+        if (keyColor.isValid()) {
+            // Draw the colored rectangle next to the key label
+            painter->fillRect(
+                    option.rect.x(),
+                    option.rect.y() + 2,
+                    4, // width
+                    option.rect.height() - 4,
+                    keyColor);
+        }
+    }
+
+    QString keyText = KeyUtils::keyToString(key);
+    if (QString("INVALID").compare(keyText) == 0) { // if the key is invalid, display empty string
```

Comment:
not the other way around, since we already have a QString and don't need to construct another one?
(based what I think I understood fo QString stuff)
```suggestion
    if (keyText.compare("INVALID") == 0) { // if the key is invalid, display empty string
```

Reply:
good suggestion, but better to avoid string comparison altogether. https://github.com/mixxxdj/mixxx/pull/13390#discussion_r1672347393